src/CacheManager.StackExchange.Redis/RedisCacheHandle.cs

```diff
@@ -270,6 +270,19 @@ public override bool Exists(string key, string region)
         }
 
         /// <inheritdoc />
+        public override IEnumerable<string> Keys(string pattern, string region)
+        {
+            var keyPattern = GetKey(pattern, region);
+
+            // Keys are spread out across the cluster, slaves should contain a complete copy of their master nodes, so ignore them.
+            return _connection
+                .Servers
```

Comment:
We talked about checking for slave servers. The thing is, usually you want to have expensive reads, like `Scan` or `Keys` commands run explicitly ON slaves only, to not stress the master servers...
Either a configuration option again or more logic into here to check if there are slaves, and if so, use them instead...